src/julia/tools.py

```diff
@@ -140,13 +140,31 @@ def julia_py_executable():
     """
     Path to ``julia-py`` executable installed for this Python executable.
     """
-    scripts_path = sysconfig.get_path("scripts")
-    stempath = os.path.join(scripts_path, "julia-py")
-    candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
+
+    # try to find installed julia-py script - check scripts folders under different installation schemes
+    # we check the alternate schemes first, at most one of which should give us a julia-py script
+    # see https://docs.python.org/3/install/index.html#alternate-installation
+    scripts_paths_failed = []
+    for scheme in sysconfig.get_scheme_names():
+        scripts_path = sysconfig.get_path("scripts", scheme)
+        stempath = os.path.join(scripts_path, "julia-py")
+        candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
+        if candidates:
+            break
+        else:
+            scripts_paths_failed.append(scripts_path)
+
+    # try standard install scripts path if nothing found in alternate scheme
+    if not candidates:
+        scripts_path = sysconfig.get_path("scripts")
+        stempath = os.path.join(scripts_path, "julia-py")
+        candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
```

Comment:
Let's factor it out as a function `_julia_py_candidates`.

Reply:
Actually, I think it'd be better to do

```python
scripts_paths = [sysconfig.get_path("scripts", scheme) for scheme in sysconfig.get_scheme_names()]
scripts_paths.append(sysconfig.get_path("scripts"))
```

and loop over `scripts_paths`.

Reply:
OK yes, have collapsed the two sections into one as suggested.